google-account-plugin/src/com/google/cloud/tools/intellij/login/IntellijGoogleLoginService.java

```diff
@@ -223,8 +224,19 @@ public void onLoginCompleted() {
           };
           users.addUser(new CredentialedUser(state, localCallback));
         }
-        else if (callback != null) {
-          callback.onLoginCompleted();
+        else {
+          restorePreviousUser();
```

Comment:
could be helpful to add a comment here that this is for the case that the login failed or was cancelled